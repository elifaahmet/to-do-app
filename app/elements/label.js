import React from 'react';
import styled from 'styled-components/native';

import { themeStylesToCssString } from '../utils/helpers';

const StyledText = styled.Text`
    ${() => themeStylesToCssString('header-0')};  
    text-align: center;
` 
export const Label = (props) => {
    return <StyledText {...props}>{props.children}</StyledText>
};

export default Label;