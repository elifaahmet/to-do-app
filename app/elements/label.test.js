import * as React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import { Label } from './label';

describe('Testing Label Component', () => {
  test('it should be rendered', async () => {
    const component = (
      <Label>
        Testing
      </Label>
    );

    const { getByText, toJSON } = render(component);
    
    const element = getByText('Testing');
    expect(element.children.length).toEqual(1);

    expect(toJSON()).toMatchSnapshot();
  });

});