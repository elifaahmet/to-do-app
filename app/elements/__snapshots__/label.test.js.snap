// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Label Component it should be rendered 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Poppins_400Regular",
        "fontSize": 18,
        "fontWeight": "600",
        "lineHeight": 28,
        "textAlign": "center",
      },
    ]
  }
>
  Testing
</Text>
`;
